src/meta/processors/zoneMan/DropZoneProcessor.cpp

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/zoneMan/DropZoneProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void DropZoneProcessor::process(const cpp2::DropZoneReq& req) {
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::zoneLock());
+    auto zoneName = req.get_zone_name();
+    auto zoneIdRet = getZoneId(zoneName);
+    if (!zoneIdRet.ok()) {
+        LOG(ERROR) << "Zone " << zoneName << " not found";
+        handleErrorCode(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+
+    std::vector<std::string> keys;
+    keys.emplace_back(MetaServiceUtils::indexZoneKey(zoneName));
+    keys.emplace_back(MetaServiceUtils::zoneKey(zoneName));
+    LOG(INFO) << "Drop Zone " << zoneName;
+    doSyncMultiRemoveAndUpdate(std::move(keys));
```

Comment:
Need to check all hosts in zone is empty?

Reply:
maybe we should check whether the zone belong to any group and the group is related to a space?